### PR TITLE
LPS-201886 Confirm before deleting

### PR DESCRIPTION
**Pull request by brookedalton**
https://liferay.atlassian.net/browse/LPS-201886

### CODE REVIEW

File: modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/PublicationsDisplayContext.java [hunk 1]
```diff
@@ -455,7 +455,7 @@ public List<FDSActionDropdownItem> getFDSActionDropdownItems() {
 				).buildString(),
 				"times-circle", "delete",
 				_language.get(_httpServletRequest, "delete"), "get", "delete",
-				null),
+				"async"),
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/PublicationsDisplayContext.java:458]: This reverts your fix in https://github.com/liferay/liferay-portal/commit/61e69aa8dac8bc4690a5cb012b54c96c222c0587 .

Please resend with a solution that takes care of both issues.

**Review by pei-jung — CHANGES_REQUESTED** [review]

File: modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/view_publications.jsp [hunk 2]
```diff
@@ -17,6 +17,7 @@
 		creationMenu="<%= publicationsDisplayContext.getCreationMenu() %>"
 		fdsActionDropdownItems="<%= publicationsDisplayContext.getFDSActionDropdownItems() %>"
 		id="<%= PublicationsFDSNames.PUBLICATIONS_ONGOING %>"
+		propsTransformer="publications/js/DeletePublicationsFDSPropsTransformer"
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/view_publications.jsp:20]: Move the js file to the `js/components` directory.

File: modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/js/DeletePublicationsFDSPropsTransformer.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {fetch, openConfirmModal, openToast} from 'frontend-js-web';
+
+export default function propsTransformer({...otherProps}) {
+	const handleDelete = (isConfirmed, itemData) => {
+		const deleteURL = itemData.actions.delete.href;
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/js/DeletePublicationsFDSPropsTransformer.js:10]: Declare var close to its usage.

File: modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/js/DeletePublicationsFDSPropsTransformer.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+		if (isConfirmed) {
+			fetch(deleteURL.replace('{id}', itemData.id), {method: 'DELETE'})
+			.then((response) => {
+				if (response.ok) {
+					window.location.reload();
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/js/DeletePublicationsFDSPropsTransformer.js:16]: Add line break

File: modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/js/DeletePublicationsFDSPropsTransformer.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+					openToast({
+						message: Liferay.Language.get(
+							'your-request-completed-successfully'
+						),
+						type: 'success',
+					});
+					}
+					else {
+						throw new Error();
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/js/DeletePublicationsFDSPropsTransformer.js:25]: Use `openToast` to display the error

**Review by pei-jung — CHANGES_REQUESTED** [review]

**Comment by brookedalton** [comment]
ci:test:sf

**Review by pei-jung — APPROVED** [review]
@brookedalton good job figuring out a solution for this! 👍

**Comment by pei-jung** [comment]
ci:test:publications

**Comment by pei-jung** [comment]
ci:test:relevant

**Comment by suilin** [comment]
Hi @brookedalton @pei-jung, I found two issues for current changes, I updated the issues as follows can you look at them?

- Invited Viewer users can delete the publication
1. Add a new user
2. Enalbe Publications
3. Add a new publication
4. Add some contents
5. Invited the user as a Viewer
6. Login with the new user
Actual result: The user not only has permission to view a publication but also has permission to delete it.
Expected Result: The user should only have permission to view the publication.


- Sandbox publication name contains the wrong time after deleting
1. Add a new user with a Publications User role
2. Enable Publications and Sandbox Only
3. Add a new publication
4. Login with the new user
5. The sandbox publication shows up
6. Delete the publication
Actual result: The new sandbox publication shows up but the name's time is a bit earlier than the current UTC time. Such as the publication name should be `usersn - 11/29/23 10:47` but  `usersn - 11/29/23 10:46`. I also found that after deleting the publication, the alert message shows up before reloading the page to show the publication is gone. Previously the alert message shows up after showing the publication is gone
Expected Result: The new sandbox publication shows up and the name time is the same as the current UTC time.

**Comment by pei-jung** [comment]
Resent with fixes https://github.com/liferay-publications/liferay-portal/pull/878

**Pull request closed without merging**